### DISCUSSION TITLE
Lesson 7: When is the run() getting called within Interactions.s.sol?

### DISCUSSION THREAD

**Discussion by shrutic11** [Cyfrin/foundry-full-course-cu#411] [Q&A]
Hi,

I tried a lot using console.log but never saw anything on the screen for the run() functions that we have in our Interactions.s.sol.

Also, how are we supposed to send the --broadcast using the interactions script? 

 noticed that I didn't have the broadcast folder created in my file explorer. This means that the most recently deployed contract was never getting grabbed. 

But since we are never calling the run() function in our tests, we never see that as an error. Only when you add a console statement in your run(), you realise it was never getting called.

And since it is never getting called during the test, most of the people are getting EVM: OutOfFund error. This error gets resolved when we add vm.startBroadcast () in our fundFundMe and withdrawFundMe functions. 

But adding vm.startBroadcast is not letting the vm.prank() user call the fundMe contract, since they aren't compatible together.

I want to understand, if I want to test using a test user created with vm.prank() can call the fundMe contract, what should be my approach? I don't want to test the withdraw scenario here, only the fundme.

Interactions.s.sol
```solidity
//SPDX-License-Identifier : MIT

pragma solidity 0.8.18;

import {DevOpsTools} from "foundry-devops/src/DevOpsTools.sol";
import {FundMe} from "../src/FundMe.sol";
import {Script, console} from "lib/forge-std/src/Script.sol";

contract FundFundMe is Script {
    uint256 public constant SEND_ETH_VALUE = 0.01 ether;

    function run() external {
        address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment(
            "FundMe",
            block.chainid
        );
        vm.startBroadcast();
        fundFundMe(mostRecentlyDeployed);
        vm.stopBroadcast();
    }
    }

    function fundFundMe(address mostRecentlyDeployed) public {
        vm.startBroadcast();
        FundMe(payable(mostRecentlyDeployed)).sendUSD{value: SEND_ETH_VALUE}();
        vm.stopBroadcast();
    }
}

contract WithdrawFundMe is Script {
    function withdrawFundMe(address mostRecentlyDeployed) public {
        vm.startBroadcast();
        FundMe(payable(mostRecentlyDeployed)).withdraw();
        vm.stopBroadcast();
    }

    function run() external {
        address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment(
            "FundMe",
            block.chainid
        );
        vm.startBroadcast();
        withdrawFundMe(mostRecentlyDeployed);
        vm.stopBroadcast();
    }
}
```

**Comment by shikhar229169** [comment]
So, if you want to see the console.log stuff for the run function, you need to provide at least 2 verbosity which is -vv at the end of command.
Also in order to get the broadcast folder you need to first deploy the FundMe contract in any network either local network (anvil) or any test network. Also ensure that you run the interactions on the same network you deployed your contracts on. For DevOpsTools to work, you need to add ```ffi=true``` in foundry.toml file. 
Even after doing this if the DevOpsTool throws error like no contract deployed, then run ```foundryup``` and try again.
If you have not installed ```jq``` which is required for DevOpsTool, you can install it by:
```
sudo apt install jq
```
